coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -105,27 +106,64 @@ class MultiPartitionPlanner(partitionLocationProvider: PartitionLocationProvider
                                    qContext: QueryContext,
                                    forceInProcess: Boolean = false): PlanResult = {
     // Should avoid this asInstanceOf, far many places where we do this now.
-    val params = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
     // MultiPartitionPlanner has capability to stitch across time partitions, however, the logic is mostly broken
     // and not well tested. The logic below would not work well for any kind of subquery since their actual
     // start and ends are different from the start/end parameter of the query context. If we are to implement
     // stitching across time, we need to to pass proper parameters to getPartitions() call
-    val paramToCheckPartitions = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
-    val partitions = getPartitions(logicalPlan, paramToCheckPartitions)
-
-    if (isSinglePartition(partitions)) {
-
-      val (partitionName, startMs, endMs, grpcEndpoint) = partitions.headOption match {
+    if (forceInProcess) {
+      // If inprocess is required, we will rely on the DefaultPlanner's implementation as the expectation is that the
+      // raw series is doing a remote call to get all the data.
+      logicalPlan match {
+        case lp: RawSeries    if lp.supportsRemoteDataCall=>
+            val params = qContext.origQueryParams.asInstanceOf[PromQlQueryParams]
+            val rs = lp.rangeSelector.asInstanceOf[IntervalSelector]
+
+            val (rawExportStart, rawExportEnd) =
+              (rs.from - lp.offsetMs.getOrElse(0L) - lp.lookbackMs.getOrElse(0L), rs.to - lp.offsetMs.getOrElse(0L))
+
+            val partition = getPartitions(lp, params)
+            // For each partition, do a raw data export range query
+            val execPlans = partition.map(pa => {
+              val (thisPartitionStartMs, thisPartitionEndMs) =
+                (Math.max(pa.timeRange.startMs, rawExportStart), Math.min(pa.timeRange.endMs, rawExportEnd))
+              val totalOffsetThisPartitionMs = thisPartitionEndMs - thisPartitionStartMs
+              val thisPartitionLp = lp.copy(offsetMs = None, lookbackMs = Some(totalOffsetThisPartitionMs))
+              val newPromQlParams = params.copy(promQl = LogicalPlanParser.convertToQuery(thisPartitionLp),
+                  startSecs = thisPartitionEndMs / 1000L,
+                  endSecs = thisPartitionEndMs / 1000L,
+                  stepSecs = 1
+                )
+              val newContext = qContext.copy(origQueryParams = newPromQlParams)
+              materializeForPartition(thisPartitionLp, pa, newContext)
+            })
+          PlanResult(
+            Seq( if (execPlans.tail == Seq.empty) execPlans.head
```

Comment:
We can the partition assignments is non empty, that should be enough.  The reason I didn't put that assertion is because this logic to export raw data from remote partitions is in invoked in the context where there are at least two partitions involved.